38-audio-classifier/README.md

```diff
@@ -0,0 +1,220 @@
+
+### `detections.qo`
+
+Detection notes are published to the `detections.qo` Notefile:
+
+```json
+{
+    "body":
+    {
+        "label": "faucet",
+        "probability": 0.96
+    }
+}
+```
+
+Here, the model predicted that the faucet was running with a probability of 96%.
+
+## Classifier Firmware
+
+With the model trained and Notehub configured, it's time to build and flash the classifier firmware onto the Swan.
+
+1. From the command line, pull in the [notecard-env-var-manager](https://github.com/blues/notecard-env-var-manager) submodule that the firmware depends on:
+    ```sh
+    $ git submodule update --init 38-audio-classifier/classifier/lib/notecard_env_var_manager
+    ```
+1. Use VS Code to open the directory `38-audio-classifier/classifier`.
+1. In the file explorer, open `main.cpp` and uncomment this line: `// #define PRODUCT_UID "com.my-company.my-name:my-project"`. Replace `com.my-company.my-name:my-project` with the [ProductUID of the Notehub project](https://dev.blues.io/notehub/notehub-walkthrough/#finding-a-productuid) you created in [Notehub Setup](#notehub-setup).
+1. You will also need to add the header file for your classifier in `main.cpp`:
+    ```c
+    // Include your specific Edge Impulse header file here:
+    // #include <running_faucet_detector_inferencing.h>
+    ```
+    Uncomment the `#include` and change `running_faucet_detector_inferencing.h` to the name of your specific header file. This header file is located in the `src` subdirectory of the directory you unzipped in [Downloading the Model](#downloading-the-model).
+1. Click the PlatformIO icon, and under the Project Tasks menu, click "Build" to build the firmware image.
+1. Prepare the Swan to receive the firmware image by doing the [BOOT + RST button press combo](https://dev.blues.io/quickstart/swan-quickstart/#programming-swan-the-stlink-v3mini).
+1. Under the Project Tasks menu, click "Upload" to upload the firmware image to the Swan.
+1. Under the Project Tasks menu, click "Monitor" to view the serial log. You should see output like this:
+
+```
+{"body":{"label":"faucet","detection_threshold":"0.7","publish_rate":"60"}}
+Label of interest set to faucet.
+Detection threshold set to 0.70.
+Publish rate set to 60 seconds.
+(DSP: 276 ms., Classification: 25 ms.)
+    faucet: 0.00
+    noise: 1.00
+(DSP: 276 ms., Classification: 25 ms.)
+    faucet: 0.00
+    noise: 1.00
+.
+.
+.
+```
+
+This type of message
+
+```
+(DSP: 276 ms., Classification: 25 ms.)
+    faucet: 0.00
+    noise: 1.00
+```
+
+is purely diagnostic and indicates how long the digital signal processing (DSP) and classification operations took, in milliseconds. Then, it shows the probabilities output by the model for each class.
+
+## Testing
+
+Move your laptop and the hardware so that the microphone is in range of the sound of a faucet. Start the Monitor task in VS Code so you can see the serial log. Turn on the faucet, and you should see this in the log, if detection was successful:
```

Comment:
```suggestion
Move your laptop and the hardware so that the microphone is in range of the sound of a faucet. Start the Monitor task in VS Code so you can see the serial log. Turn on the faucet, and you should see this in the log if detection was successful:
```